utils/fsnotify.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/cri-o/cri-o/internal/log"
+	"github.com/fsnotify/fsnotify"
+)
+
+func Watch(ctx context.Context, path string, opsToWatch []fsnotify.Op) chan error {
+	externalErrorCh := make(chan error)
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		externalErrorCh <- fmt.Errorf("failed to create new watch: %v", err)
+		return externalErrorCh
+	}
+	defer watcher.Close()
+
+	done := make(chan struct{})
+	doneClosed := false
+	internalErrorCh := make(chan error)
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				log.Debugf(ctx, "Event: %v", event.Op)
+				for _, op := range opsToWatch {
+					if event.Op&op == op {
+						if event.Name == path {
+							done <- struct{}{}
+							return
+						}
+					}
+				}
+			case err := <-watcher.Errors:
+				if err != nil {
+					internalErrorCh <- fmt.Errorf("watch error %v", err)
```

Comment:
I would wrap those errors:
```suggestion
					internalErrorCh <- errors.Wrap(err, "watch failed")
```